src/app/themes.tsx

```diff
@@ -10,52 +10,112 @@ import {
     createTheme as createMaterialTheme,
 } from "@mui/material";
 
-type CustomTheme = {
-    foregroundColor: string;
-    backgroundColor: string;
-    primaryForegroundColor: string;
-    primaryBackgroundColor: string;
-    secondaryForegroundColor: string;
-    secondaryBackgroundColor: string;
-    accentBackgroundColor: string;
-    accentForegroundColor: string;
-    surfaceForegroundColor: string;
-    surfaceBackgroundColor: string;
-    errorColor: string;
+export type standardPalette = {
+    background: string[];
+    foreground: string[];
+    largeForeground: string[];
 };
 
-const lightTheme: CustomTheme = {
-    foregroundColor: "#000000",
-    backgroundColor: "#eeeeee",
-    primaryForegroundColor: "#000000",
-    primaryBackgroundColor: "#63a036",
-    secondaryForegroundColor: "#ffffff",
-    secondaryBackgroundColor: "#066940",
-    accentBackgroundColor: "#609fd3",
-    accentForegroundColor: "#000000",
-    surfaceForegroundColor: "#000000",
-    surfaceBackgroundColor: "#ffffff",
-    errorColor: "#ff624e",
+export type rainbowPalette = {
+    red: standardPalette;
+    orange: standardPalette;
+    yellow: standardPalette;
+    green: standardPalette;
+    blue: standardPalette;
+    purple: standardPalette;
+    grey: standardPalette;
+    brown: standardPalette;
 };
 
-const darkTheme: CustomTheme = {
-    foregroundColor: "#ffffff",
-    backgroundColor: "#1a1a1a",
-    primaryForegroundColor: "#ffffff",
-    primaryBackgroundColor: "#066940",
-    secondaryForegroundColor: "#000000",
-    secondaryBackgroundColor: "#63a036",
-    accentBackgroundColor: "#1b629c",
-    accentForegroundColor: "#ffffff",
-    surfaceForegroundColor: "#ffffff",
-    surfaceBackgroundColor: "#161414",
-    errorColor: "#ff624e",
+export type customTheme = {
+    light: boolean;
+    main: standardPalette & { lighterForeground: string; border: string };
+    primary: standardPalette;
+    error: string;
+    shadow: string;
+    rainbow: rainbowPalette;
 };
 
-interface Props {
-    children: React.ReactElement;
-    theme?: CustomTheme;
-}
+const BLACK = "#000000";
+const WHITE = "#f2f2f2";
+
+export const lightTheme: customTheme = {
+    light: true,
+    main: {
+        background: ["#fdfdfd", "#eeeeee"],
+        foreground: [BLACK, "#2d2d2d"],
+        largeForeground: [BLACK, "#2d2d2d"],
+        lighterForeground: "#666666",
+        border: "#d5d5d5",
+    },
+    primary: {
+        background: ["#a3e0b0", "#7fd495", "#02985a", "#05663f", "#154d30"],
+        foreground: [BLACK, BLACK, BLACK, WHITE, WHITE],
+        largeForeground: [BLACK, BLACK, WHITE, WHITE, WHITE],
+    },
+    error: "#ff624e",
+    shadow: "#e2e2e2",
+    rainbow: {
+        red: {
```

Comment:
Could we use a `lightX` and `darkX` instead of an array here?

Reply:
Done